cbt_export_changes.py

```diff
@@ -74,13 +73,17 @@ def save_changed_blocks(changed_blocks, output_file):
             out.write(b)
 
 
-def download_changed_blocks(bitmap, uri, changed_blocks_output_path,
+def download_changed_blocks(bitmap, get_nbd_info, changed_blocks_output_path,
                             bitmap_output_path):
 
     print "downloading changed blocks"
-    # Extract the host ip address from the URI
-    host = uri.split("/")[2].split(":")[0]
-    blocks = get_changed_blocks(host, uri, bitmap, bitmap_output_path)
+    cert_text = get_nbd_info['cert']
+    with open(certfile, 'w') as cert_out:
+        cert_out.write(cert_text)

```

Comment:
Tested again to see if I could get the client to work with a string rather than a file but still didn't look like it would work (error message indicated that a file is expected). Spoke to Gabor who said he didn't do anything fancy to get it working with the python 3 version of the client so I think it may just be a limitation of python2.7

Reply:
If you check the certificates (or public keys) are the same (as described in another comment) then you won't need CA certificates, so you won't need to write this to a file.